graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/modules/PosixModuleBuiltins.java

```diff
@@ -1691,11 +1691,9 @@ private Object rename(String src, String dst) {
         @Specialization
         @TruffleBoundary(allowInlining = true)
         PBytes urandom(int size) {
-            // size is in bytes
-            BigInteger bigInteger = new BigInteger(size * 8, new Random());
-            // sign may introduce an extra byte
-            byte[] range = Arrays.copyOfRange(bigInteger.toByteArray(), 0, size);
-            return factory().createBytes(range);
+            byte[] bytes = new byte[size];
+            new Random().nextBytes(bytes);
```

Comment:
nit: this is ok because graal used `Random` in their code, so this code works the same and fix the bug, but I'm looking at the docs for `urandom` (https://docs.python.org/3/library/os.html) and they say:
> Return a string of size random bytes suitable for cryptographic use.

but java `Random` is not considered cryptographically secure. 
another option is to use `SecureRandom` (it also has `nextBytes` so the rest is the same).

Reply:
They also say this in the docs:
> The returned data should be unpredictable enough for cryptographic applications, though its exact quality depends on the OS implementation.

so maybe `Random` is ok, I'm assuming there is a library in python that is better for cryptographic stuff.

Reply:
Hmm...

I think you may be right. Based on the description of urandom, the description of SecureRandom, and the source code of normal Random, it seems like SecureRandom is a better match.

That said, moving from Random to SecureRandom is a much bigger change than fixing this bug, and not something I think I can say with certainty is the right thing to do. I'll add it as a comment on my open PR to upstream the change and see if anyone over there has feelings about it.